src/components/searchArea/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, {useCallback} from 'react'
+import {FiSearch} from 'react-icons/fi'
+
+import { Container } from './styles'
+
+const SearchArea: React.FC = () => {
+  const handleWithSearch = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+
+    const list = document.querySelectorAll('.css-1acecyi span')
+
+    if(value !== ''){
+      list.forEach(item => {
+        const regex = new RegExp(value, 'gi')
+        const text = item.textContent as string
+        const has = text.match(regex)
+  
+        if (has){
+          item.id = "red"
+        } else {
+          item.id = ""
+        }
+      })
+    } else {
+      list.forEach(item => item.id = "")
+    }
+  }, [])
+
+  return (
+    <Container>
+      <input
+        placeholder="Search..."
+        onChange={event => handleWithSearch(event)}
```

Comment:
Also, this value needs to be debounced, because the json might be huge, and searching on every keypress would make the search operation too slow in some scenarios.

Reply:
Of course, I didn't think about the performance. I'll fix it too 😅